### PR TITLE
Load tools only once at the startup

### PR DESCRIPTION
**Pull request by angrybayblade**

### CODE REVIEW

File: packages/valory/skills/task_execution_abci/behaviours.py [hunk 1]
```diff
@@ -128,7 +128,11 @@ def async_act(self) -> Generator:  # pylint: disable=R0914,R0915
                     self.context.logger.error(
                         f"Failed to get the tools {tools} with file_hash {file_hash} from IPFS!"
                     )
-                all_tools.update({tool: tool_py for tool in tools})
+                local_namespace: Dict[str, Any] = globals().copy()
+                if "run" in local_namespace:
+                    del local_namespace["run"]
+                exec(str(tool_py), local_namespace)  # pylint: disable=W0122  # nosec
+                all_tools.update({tool: local_namespace["run"] for tool in tools})
```
Comment (0xArdi) [inline packages/valory/skills/task_execution_abci/behaviours.py:135]: did you try running this?

Comment (0xArdi) [reply packages/valory/skills/task_execution_abci/behaviours.py:135]: also would be good to have these changes here: https://github.com/valory-xyz/mech/pull/103

Comment (angrybayblade) [reply packages/valory/skills/task_execution_abci/behaviours.py:135]: I'll test this using a script locally

**Review by jmoreira-valory — APPROVED** [review]
Reviewed the tool loading refactoring

**Pull request merged**